include/gameobject.h

```diff
@@ -14,33 +15,52 @@ namespace nautical
         // World Transformation for GameObject
         Transform<float> transform;
 
-        GameObject(char* name);
-
-        virtual ~GameObject();
-
         // Init Function (called at object creation)
-        virtual void init();
+        void init();
```

Comment:
No reason to ever override game objects. We override script objects to their language specific sub-classes.